### PR TITLE
[Proof of concept] Clicking on nodes to check information

### PR DESCRIPTION
**Pull request by sultansidhu**

### CODE REVIEW

File: backend/server.py [hunk 1]
```diff
@@ -34,10 +39,33 @@ def get_tree_by_id(id):
...
+        """
+        Gets the associated link from the database, gets the text associated
+        with it, and summarizes it.
+        """
+
+        def _get_content_from_site(url):
+            page = requests.get(url)
+            soup = BeautifulSoup(page.content, "html.parser")
+            content = soup.find_all(["h1", "h2", "p"])
+            return "\n".join([c.text for c in content])
```
Comment (slakh96) [inline backend/server.py]: Is there any better way to format this? In the frontend I can't see any paragraphs and the article is difficult to read due to this

File: frontend/src/components/InfoWindow/InfoWindow.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * The react component for the information window for each node
+ * in the concept tree.
+ */
+
+// handle right clicks with
+// https://stackoverflow.com/questions/31110184/react-synthetic-event-distinguish-left-and-right-click-events
```
Comment (slakh96) [inline frontend/src/components/InfoWindow/InfoWindow.jsx]: Should we move this link into the ticket description instead?

**Comment by slakh96** [comment]
Yikes we forgot about unittesting the backend

File: backend/repopulate_db.py [hunk 3]
```diff
@@ -2,25 +2,38 @@
...
     deleted_data = radialTrees.delete_many({})
+    deleted_links = nodeLinks.delete_many({})
 
     # Tree for testing purposes
     tree1 = {
         "id": 1,
         "data": {"name": "Root", "children": [{"name": "Front", "children": []}]},
     }
 
+    # Links for testing purposes
```
Comment (rafiiary) [inline backend/repopulate_db.py]: Why are we merging links for testing purposes into the main codebase?

File: backend/repopulate_db.py [hunk 4]
```diff
@@ -374,3 +387,10 @@
     # Insert another tree
     result = radialTrees.insert_one(tree2)
     print(result.inserted_id)
+    # Links for testing purposes
```
Comment (rafiiary) [inline backend/repopulate_db.py]: Same thing as above here

File: frontend/src/components/InfoWindow/InfoWindow.jsx (hunk 2 above)
Comment (rafiiary) [reply frontend/src/components/InfoWindow/InfoWindow.jsx]: Why do we even have this link?

File: frontend/src/utils/utils.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    ref.current = value;
+  });
+  return ref.current;
+}
+
+/**
+ * Replaces the spaces in a given string with hyphens.
+ * @param {string} urlString The string to be converted from spaces to hyphens.
+ */
+export function replaceSpaceCharacters(urlString) {
```
Comment (rafiiary) [inline frontend/src/utils/utils.js:20]: I think we can just use Python's own replace() and strip() methods. Won't be more expensive and will look cleaner

File: frontend/src/components/InfoWindow/InfoWindow.jsx (hunk 2 above)
Comment (slakh96) [reply frontend/src/components/InfoWindow/InfoWindow.jsx]: @sultansidhu

File: backend/repopulate_db.py (hunk 3 above)
Comment (slakh96) [reply backend/repopulate_db.py]: So these links (and this entire file in general) is solely used as a convenient way to repopulate our local databases to make it easier to develop locally; so this link is the one we read from in order to populate the side info text box when the user right clicks on a node. This file and this data will not be used at all in the actual release, it is just for us. I have updated the comments for this file to try to be more clear

File: backend/repopulate_db.py (hunk 4 above)
Comment (slakh96) [reply backend/repopulate_db.py]: https://github.com/VerWiki/VerWikiApp/pull/7#discussion_r575200411

File: frontend/src/components/InfoWindow/InfoWindow.jsx (hunk 2 above)
Comment (sultansidhu) [reply frontend/src/components/InfoWindow/InfoWindow.jsx]: @rafiiary This is just to keep it in front of our eyes when we develop for handling right clicks. Info for right clicks was kinda hard to find so i put the most helpful link on here. Should I move them to ticket description instead? might get lost there tho, but im okay either way

File: backend/server.py (hunk 1 above)
Comment (sultansidhu) [reply backend/server.py]: [UPCOMING TICKET]

File: frontend/src/utils/utils.js (hunk 5 above)
Comment (slakh96) [reply frontend/src/utils/utils.js:20]: Done

File: frontend/src/components/Tree/Tree.jsx [hunk 6]
```diff
@@ -139,28 +143,14 @@ function renderTree(dimensions, jsonData, svgRef, onNodeClick) {
...
-    ref.current = value;
-  });
-  return ref.current;
-}
-
-/**
- * todo
- * @param {*} z
+ * A function to bound the passed in value to 1 upper bound
+ * @param {*} z TODO sultan help pls
```
Comment (slakh96) [inline frontend/src/components/Tree/Tree.jsx]: @sultansidhu SOS

File: frontend/src/components/InfoWindow/InfoWindow.jsx (hunk 2 above)
Comment (rafiiary) [reply frontend/src/components/InfoWindow/InfoWindow.jsx]: @sultansidhu put it in the Road Map Goals under the links section please 
https://docs.google.com/document/d/1Ujy8Y-5NUPA0RghRK2dCsw5JEuLjSkR0J8mer6uI5GA/edit?usp=sharing

**Review by Guri67 — APPROVED** [review]

**Comment by slakh96** [comment]
We also have to fix the issue where Mit's bar at the top doesn't go across the whole width #TODO

**Review by slakh96 — CHANGES_REQUESTED** [review]
We also have to fix the issue where Mit's bar at the top doesn't go across the whole width #TODO

**Comment by slakh96** [comment]
> We also have to fix the issue where Mit's bar at the top doesn't go across the whole width #TODO

Done, but now the simrat rendering bug is back :scream: :fearful:

File: frontend/src/components/Tree/Tree.jsx (hunk 6 above)
Comment (sultansidhu) [reply frontend/src/components/Tree/Tree.jsx]: Done

**Comment by slakh96** [comment]
Whitespace on the right side of the textbox
Future ticket: Leaves not clickable, and visibly look different than non leaves

**Review by slakh96 — APPROVED** [review]

**Review by slakh96 — CHANGES_REQUESTED** [review]
Yikes the textbox is not bounded by the height of the tree anymore :/ we gotta fix that first

**Review by slakh96 — APPROVED** [review]

**Pull request merged**